apps/prairielearn/src/lib/client/safe-db-types.ts

```diff
@@ -1,11 +1,99 @@
 import { type z } from 'zod';
 
 import {
+  AssessmentInstanceSchema as RawAssessmentInstanceSchema,
+  AssessmentSchema as RawAssessmentSchema,
+  AssessmentSetSchema as RawAssessmentSetSchema,
   CourseInstanceSchema as RawCourseInstanceSchema,
   CourseSchema as RawCourseSchema,
   UserSchema as RawUserSchema,
 } from '../db-types.js';
 
+/** Assessments */
+export const RawStaffAssessmentSchema = RawAssessmentSchema;
+export const StaffAssessmentSchema = RawStaffAssessmentSchema.brand<'StaffAssessment'>();
+export type StaffAssessment = z.infer<typeof StaffAssessmentSchema>;
+
+export const RawStudentAssessmentSchema = RawStaffAssessmentSchema.pick({
+  advance_score_perc: true,
+  allow_issue_reporting: true,
+  allow_real_time_grading: true,
+  allow_personal_notes: true,
+  assessment_module_id: true,
+  assessment_set_id: true,
+  auto_close: true,
+  constant_question_value: true,
+  course_instance_id: true,
+  deleted_at: true,
```

Comment:
I agree this is not sensitive. In any case it should always be `null` when we are sending data to students. Could we enforce this somehow so it's an error if we try and convert an assessment schema with a non-null value for `deleted_at`? This is super low importance, I was just idly wondering.